boot/freeldr/freeldr/arch/i386/xboxvideo.c

```diff
@@ -135,35 +133,17 @@ XboxVideoInit(VOID)
   /* FIXME: obtain fb size from firmware somehow (Cromwell reserves high 4 MB of RAM) */
   FrameBufferSize = 4 * 1024 * 1024;
 
-  /* FIXME: don't use SMBus, obtain current video resolution directly from NV2A */
-  if (I2CTransmitByteGetReturn(0x10, 0x04, &AvMode))
-    {
-      if (1 == AvMode) /* HDTV */
-        {
-          ScreenWidth = 720;
-        }
-      else
-        {
-          /* FIXME Other possible values of AvMode:
-           * 0 - AV_SCART_RGB
-           * 2 - AV_VGA_SOG
-           * 4 - AV_SVIDEO
-           * 6 - AV_COMPOSITE
-           * 7 - AV_VGA
-           * other AV_COMPOSITE
-           */
-          ScreenWidth = 640;
-        }
-    }
-  else
-    {
-      ScreenWidth = 640;
-    }
-
-  ScreenHeight = 480;
+  ScreenWidth = *((PULONG) 0xfd680838) + 1; /* NV_RAMDAC_FP_HVALID_END */
```

Comment:
Why don't you define these constants?

Reply:
I tried to define them in `machxbox.h` like that:
```c
#define NV2A_CONTROL_OFFSET            0xFD000000
#define   NV2A_CRTC_OFFSET               0x600000 + NV2A_CONTROL_OFFSET
#define     NV2A_CRTC_FRAMEBUFFER_START     0x800 + NV2A_CRTC_OFFSET
#define   NV2A_RAMDAC_OFFSET             0x680000 + NV2A_CONTROL_OFFSET
#define     NV2A_RAMDAC_FP_HVALID_END       0x838 + NV2A_RAMDAC_OFFSET
#define     NV2A_RAMDAC_FP_VVALID_END       0x818 + NV2A_RAMDAC_OFFSET
```
But once it's done, I'm getting zero values into variables. I checked with DPRINT -- the addresses are correct. Any idea why it happens? :thinking:

Reply:
Place the values within parentheses. E.g.
`#define   NV2A_CRTC_OFFSET               (0x600000 + NV2A_CONTROL_OFFSET)`

Reply:
Thanks, it worked! Fixed. :ok_hand: